### PR TITLE
fix sync error

### PR DESCRIPTION
**Pull request by blagoev**
related docs fix https://github.com/realm/realm.io/pull/3876

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -361,7 +361,14 @@ declare namespace Realm.Sync {
         expiresAt?: Date;
     }
 
-    type ErrorCallback = (message?: string, isFatal?: boolean, category?: string, code?: number) => void;
+    class SyncError {
```
Comment (fealebenpae) [inline lib/index.d.ts]: I think this should be just an interface.

File: lib/index.d.ts [hunk 2]
```diff
@@ -361,7 +361,14 @@ declare namespace Realm.Sync {
...
+        message: string; 
+        isFatal: boolean 
+        category?: string 
+        code: number;
+    }
+    
+    type ErrorCallback = (sender, error: SyncError) => void;
```
Comment (fealebenpae) [inline lib/index.d.ts]: `sender` should be `sender: Session`, I think.

Comment (blagoev) [reply lib/index.d.ts]: thanks, if we are sure this is always the case I can do it. I was thinking to do it `sender: any` and then i missed to do it. 
Why not call it `session: Session` then

File: lib/index.d.ts (hunk 1 above)
Comment (blagoev) [reply lib/index.d.ts]: right will do

File: lib/index.d.ts [hunk 3]
```diff
@@ -361,7 +361,14 @@ declare namespace Realm.Sync {
         expiresAt?: Date;
     }
 
-    type ErrorCallback = (message?: string, isFatal?: boolean, category?: string, code?: number) => void;
+    interface SyncError {
+        message: string;
```
Comment (kneth) [inline lib/index.d.ts]: Is only `category` optional?

Comment (blagoev) [reply lib/index.d.ts]: i think so. from the code it seems all other will be present with some value

**Pull request merged**